lib/utils.pm

```diff
@@ -85,7 +85,8 @@ sub wait_boot {
         }
         return;
     }
-    else {
+    # On Xen PV we don't see a Grub menu
+    elsif (!(check_var('VIRSH_VMM_FAMILY', 'xen') && check_var('VIRSH_VMM_TYPE', 'linux'))) {
```

Comment:
what about keeping an explicit else and failing there with a "missing implementation" or something